src/components/ThirdPartyCollectionDetailPage/CollectionPublishButton/CollectionPublishButton.tsx

```diff
@@ -5,26 +5,83 @@ import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 import { Button } from 'decentraland-ui'
 import { env } from 'decentraland-commons'
 import { MAX_PUBLISH_ITEM_COUNT } from 'modules/thirdParty/utils'
+import { getCollectionType } from 'modules/collection/utils'
+import { CollectionType } from 'modules/collection/types'
+import { SyncStatus } from 'modules/item/types'
 import UnderReview from './UnderReview'
 import { Props } from './CollectionPublishButton.types'
 
+enum ButtonAction {
+  PUBLISH,
+  PUSH_CHANGES,
+  PUBLISH_AND_PUSH_CHANGES
+}
+
 const CollectionPublishButton = (props: Props) => {
-  const { collection, items, slots, onPublish } = props
+  const { collection, items, slots, onPublish, onPushChanges, itemsStatus } = props
+  const isTP = getCollectionType(collection) === CollectionType.THIRD_PARTY
 
   const isPublishDisabled = useMemo(() => {
     return (
       !env.get('REACT_APP_FF_WEARABLES_PUBLISH') ||
       slots === 0 ||
       items.length === 0 ||
-      items.length > MAX_PUBLISH_ITEM_COUNT ||
-      collection.isPublished
+      (!isTP && items.length > MAX_PUBLISH_ITEM_COUNT) ||
+      (!isTP && collection.isPublished)
     )
-  }, [items])
+  }, [items, slots, collection])
+
+  const buttonAction = useMemo(() => {
+    let action = ButtonAction.PUBLISH
+    const { willPublish, willPushChanges } = Object.values(itemsStatus).reduce(
+      (acc, status) => {
+        if (status === SyncStatus.UNPUBLISHED) {
+          acc.willPublish = true
+        } else if (status === SyncStatus.UNDER_REVIEW) {
+          acc.willPushChanges = true
+        }
+        return acc
+      },
+      {
+        willPublish: false,
+        willPushChanges: false
+      }
+    )
```

Comment:
That was exactly what I wrote the first time. But then I was like "I shouldn't iterate over the entire array twice, once should be enough". So that's why I wrote a reduce to just go through the entire array one time and get both flags defined. 
What do you think? I can go back to the `.some` if you want. I know it's more code and maybe uglier :p but I believe it will perform better, specially having huge arrays as we probably will.